### PR TITLE
Redefine CSS literal rules for CSS in JS

### PR DESCRIPTION
**Pull request by black7375**
This is the first RFC.
I've write it easy to see how it transforms.

- [Rendered](https://github.com/mincho-js/working-group/blob/css-literals/text/000-css-literals.md)

---

Let's talk more about this tomorrow.

cc @Jeong-jj

### CODE REVIEW

File: text/000-css-literals.md [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+Inspired by the [Less's Merge properties](https://lesscss.org/features/#merge-feature), this feature allows you to composition long split values.
+
+Only `$` and `_` can use special characters that don't require a string in javascript.  
+By convention in the programming world, we use `_` to represent a whitespace, so we do the same.
+
+**Code:**
+```typescript
+export const myCss = css({
+  boxShadow$: ["inset 0 0 10px #555", "0 0 20px black"],
+  transform_: ["scale(2)", "rotate(15deg)"]
```
Comment (Jeong-jj) [inline text/000-css-literals.md:140]: `$`is will be `,` and `_` is gonna be **whitespace**. right?

File: text/000-css-literals.md [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+export const myCss = css({
+  transform_: [
+    // Apply to all
+    "scale(2)",
+
+    //  Fallback style
+    ["rotate(28.64deg)", "rotate(0.5rad)"]
+  ]
+});
+```
```
Comment (Jeong-jj) [inline text/000-css-literals.md:174]: it's so usefull

File: text/000-css-literals.md [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+
+**Compiled:**
+```css
+.[FILE_NAME]_myCSS__[HASH] {
+  color: red !important;
+}
+```
+
+## 9. CSS Variables
```
Comment (Jeong-jj) [inline text/000-css-literals.md:216]: I understand this part, but i have one question.
I remember you said there is a little bit change comparing with origin of article, so i search about that in yours but i didnt find.
Could you tell me where is origin of article?

**Review by Jeong-jj — APPROVED** [review]
It's easier to understand the requirements than before👍

File: text/000-css-literals.md (hunk 1 above)
Comment (black7375) [reply text/000-css-literals.md:140]: Yes, it is.
- `$` => join with `,`
- `_` => join with ` `

File: text/000-css-literals.md (hunk 3 above)
Comment (black7375) [reply text/000-css-literals.md:216]: In the original plan, `$` at the front was reserved for selector reference.
```typescript
const classes = cssVariants({
  parent: {
    ":hover $child": {
      width: "1px"
    }
  },
  child: {
    "$parent &": {
      width: "2px"
    }
  }
});
```

However, a concession was made to treat `$SOMETHING` like a `css variable` literal syntax, and it was replaced by the letter `%`.
```typescript
const classes = cssVariants({
  parent: {
    ":hover $child": {
      width: "1px"
    }
  },
  child: {
    "%parent &": {
      width: "2px"
    }
  }
});
```

**Pull request merged**